src/crash_report.zig

```diff
@@ -159,145 +159,6 @@ pub fn compilerPanic(msg: []const u8, error_return_trace: ?*std.builtin.StackTra
     PanicSwitch.dispatch(error_return_trace, stack_ctx, msg);
 }
 
-/// Attaches a global SIGSEGV handler
-pub fn attachSegfaultHandler() void {
-    if (!debug.have_segfault_handling_support) {
-        @compileError("segfault handler not supported for this target");
-    }
-    if (builtin.os.tag == .windows) {
-        _ = os.windows.kernel32.AddVectoredExceptionHandler(0, handleSegfaultWindows);
-        return;
-    }
-    var act = os.Sigaction{
-        .handler = .{ .sigaction = handleSegfaultPosix },
-        .mask = os.empty_sigset,
-        .flags = (os.SA.SIGINFO | os.SA.RESTART | os.SA.RESETHAND),
-    };
-
-    os.sigaction(os.SIG.SEGV, &act, null);
-    os.sigaction(os.SIG.ILL, &act, null);
-    os.sigaction(os.SIG.BUS, &act, null);
-}
-
-fn handleSegfaultPosix(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const anyopaque) callconv(.C) noreturn {
-    // TODO: use alarm() here to prevent infinite loops
-    PanicSwitch.preDispatch();
```

Comment:
This code is similar to the one in `debug.zig` but not exactly duplicated, so it can't be straight up deleted. I ran into the same issue in https://github.com/ziglang/zig/pull/10615 and I think the fix is to make `debug.zig` use the implementation in `crash_report.zig`, since it is thread-safe and handles more edge cases (like double panic).

Reply:
ok, does someone know where exactly the difference is?

Reply:
The one commented on here is one, the way that the Windows sigfault handler is set up in `attachSegfaultHandler` is another, there might be more. Point is that unifying these two implementations is more complicated than just removing and/or copy-pasting code from one place to the other.

Reply:
I removed my removal for now, but reused the `updateSegfaultHandler` to avoid a further diversion of the codes.